aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -148,6 +164,46 @@ private void associateKmsKey(final AmazonWebServicesClientProxy proxy,
         logger.log(kmsKeyMessage);
     }
 
+    private void updateTags(final AmazonWebServicesClientProxy proxy,
+                            final ResourceModel model,
+                            final Map<String, String> previousTags,
+                            final Map<String, String> tags,
+                            final Logger logger) {
+        MapDifference<String, String> tagsDifference = Maps.difference(Optional.ofNullable(previousTags).orElse(new HashMap<>()),
+                Optional.ofNullable(tags).orElse(new HashMap<>()));
+        final Map<String, String> tagsToRemove = tagsDifference.entriesOnlyOnLeft();
+        final Map<String, String> tagsToAdd = tagsDifference.entriesOnlyOnRight();
+        final Map<String, String> tagsToDiffer = tagsDifference.entriesDiffering().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, tag -> tag.getValue().rightValue()));
+        final Map<String, String> tagsToUpdate = Stream.concat(tagsToAdd.entrySet().stream(), tagsToDiffer.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        try {
+            if (!tagsToRemove.isEmpty()) {
+                final List<String> tagKeys = new ArrayList<>(tagsToRemove.keySet());
+                proxy.injectCredentialsAndInvokeV2(Translator.translateToUntagLogGroupRequest(model.getLogGroupName(), tagKeys),
+                        ClientBuilder.getClient()::untagLogGroup);
+
+                final String message =
+                        String.format("%s [%s] successfully removed tags: [%s]",
+                                ResourceModel.TYPE_NAME, model.getLogGroupName(), tagKeys);
+                logger.log(message);
+            }
+            if(!tagsToUpdate.isEmpty()) {
+                proxy.injectCredentialsAndInvokeV2(Translator.translateToTagLogGroupRequest(model.getLogGroupName(), tagsToUpdate),
+                        ClientBuilder.getClient()::tagLogGroup);
+
+                final String message =
+                        String.format("%s [%s] successfully added tags: [%s]",
+                                ResourceModel.TYPE_NAME, model.getLogGroupName(), tagsToUpdate);
+                logger.log(message);
+            }
+        } catch (final ResourceNotFoundException e) {
+            throwNotFoundException(model);
+        } catch (final InvalidParameterException e) {
```

Comment:
We need to soft fail the Tag permission as well for Update.  The reason is that the Tagging change would comes from stack tags as well. The use case is that some customers don't give Tag permission to the stack.  Before this change, the Log group resource can be update successful while doing stack tag update, as this resource just ignore any tag update.  

With this change, it will now update the tag for cloudformation stack tags and it can run into permission error and fails the resource update.  Soft fail with it as same as read handler can avoid breaking change to those customers.